source/slang/lower-to-ir.cpp

```diff
@@ -1058,9 +1058,7 @@ struct ValLoweringVisitor : ValVisitor<ValLoweringVisitor, LoweredValInfo, Lower
 
     IRType* visitNamedExpressionType(NamedExpressionType* type)
     {
-        return (IRType*) getSimpleVal(context,
-            emitDeclRef(context, type->declRef,
-                context->irBuilder->getTypeKind()));
+        return (IRType*)getSimpleVal(context, dispatchType(type->GetCanonicalType()));
```

Comment:
I think this could just be `lowerType(context, type->GetCanonicalType())`, which might be cleaner so that we have fewer places directly invoking `dispatchType()`.